### ISSUE TITLE
Routes get rendered multiple times - Hot Reload - @aurelia/vite-plugin

### ISSUE THREAD

**Issue by ivanbacher** [aurelia/aurelia#2062]
# 🐛 Bug Report

- Bootstrapped new app using: `Default TypeScript Aurelia 2 App`
- Added `router-lite` package
- Added custom element and new route

`my-app.html`
```
<import from="./my-element.html"></import>

<div class="message">${message}</div>

<my-element></my-element>

<div>
  <au-viewport></au-viewport>
</div>
```
<img width="1219" alt="Screenshot 2024-10-04 at 10 40 54" src="https://github.com/user-attachments/assets/2d15b68a-10c6-4025-a7c8-4a24fa9a2de7">

When a change happens in the new route that was created: then route is added again.

<img width="1912" alt="Screenshot 2024-10-04 at 10 42 01" src="https://github.com/user-attachments/assets/fbb0480a-a9f9-4855-8539-71fcbc04f8fb">

Page needs to be reloaded to get everything back to normal. I am assuming it could have something todo with how Vite handles reloads?

Repo:
[test-app.zip](https://github.com/user-attachments/files/17255645/test-app.zip)

**Comment by bigopon** [comment]
Yes at a glance it seems this is hot reload, since the unmounting behavior isnt smart, and because of host controller involved. Thanks @ivanbacher
